rust/cubestore/cubestore/src/metastore/mod.rs

```diff
@@ -1632,18 +2052,24 @@ trait RocksTable: Debug + Send + Sync {
         K: Hash,
     {
         let row_ids = self.get_row_ids_by_index(row_key, secondary_index)?;
-
         let mut res = Vec::new();
 
         for id in row_ids {
             if let Some(row) = self.get_row(id)? {
                 res.push(row);
             } else {
-                let index = Self::indexes()
-                    .into_iter()
-                    .find(|i| i.get_id() == BaseRocksSecondaryIndex::get_id(secondary_index))
-                    .unwrap();
+                if RocksSecondaryIndex::is_ttl(secondary_index) {
+                    trace!(
+                        "Row exists in secondary index (with TTL) however missing in {:?} table: {}. Compaction problem?",
+                        self, id
+                    );
+
+                    continue;
```

Comment:
It's data consistency problem, not compaction (because of get_row_ids_by_index filter out expired ids). I think we should rebuild index at this case, or, at least put `TODO` about it here